### PR TITLE
[fix](nereids)prevent null pointer exception if datetime value overflows

### PR DESCRIPTION
**Pull request by starocean999**
date and datetime literal's plusXXX method( plusDays, plusWeeks, plusMonths) may return NullLiteral, So we should only convert the return value to date or datetime literal when it's not a NullLiteral

### CODE REVIEW

**Comment by starocean999** [comment]
run buildall

**Comment by starocean999** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/literal/DateLiteral.java [hunk 1]
```diff
@@ -336,7 +336,7 @@ protected boolean checkDate() {
     }
 
     protected static boolean isDateOutOfRange(LocalDateTime dateTime) {
-        return dateTime.isBefore(START_OF_A_DAY) || dateTime.isAfter(END_OF_A_DAY);
+        return dateTime.isBefore(START_OF_A_DAY) || dateTime.isAfter(END_OF_A_DAY) || dateTime == null;
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/literal/DateLiteral.java:339]:
```suggestion
        return dateTime == null || dateTime.isBefore(START_OF_A_DAY) || dateTime.isAfter(END_OF_A_DAY);
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/executable/TimeRoundSeries.java [hunk 2]
```diff
@@ -104,22 +104,27 @@ private static LocalDateTime getDateCeilOrFloor(DATE tag, LocalDateTime date, in
...
+                    return ((DateTimeLiteral) start.plusHours(step)).toJavaDateType();
+                case MINUTE:
+                    return ((DateTimeLiteral) start.plusMinutes(step)).toJavaDateType();
+                case SECOND:
+                    return ((DateTimeLiteral) start.plusSeconds(step)).toJavaDateType();
+                default:
+                    break;
+            }
+        } catch (Exception ex) {
+            // do nothing
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/executable/TimeRoundSeries.java:125]: exception is expensive, could we process it without try catch

**Comment by starocean999** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by LiBinfeng-01 — APPROVED** [review]

**Pull request merged**